solr/core/src/java/org/apache/solr/util/tracing/SimplePropagator.java

```diff
@@ -139,11 +147,15 @@ public SimplePropagatorSpanBuilder(ScopeManager scopeManager) {
     @Override
     public Span start() {
       if (parent != null) {
-        return new SimplePropagatorSpan(parent);
+        if (parent instanceof SimplePropagatorSpan) {
+          return (SimplePropagatorSpan) parent;
+        } else {
+          return NoopSpan.INSTANCE;
+        }
       } else if (scopeManager.activeSpan() != null) {
-        return new SimplePropagatorSpan(scopeManager.activeSpan().context());
+        return scopeManager.activeSpan();
       } else {
```

Comment:
I think this is a double-lookup in a ThreadLocal?

Reply:
yes. good catch